internal/computeagent/computeagent.proto

```diff
@@ -17,10 +18,25 @@ message AddNICInternalRequest {
 
 message AddNICInternalResponse {}
 
+message ModifyNICInternalRequest {
+    string nic_id = 1;
+    string endpoint_name = 2;
+    IovEndpointPolicyInternalSetting iov_policy_settings = 3;
+}
+
+message ModifyNICInternalResponse {}
+
+
 message DeleteNICInternalRequest {
     string container_id = 1;
     string nic_id = 2;
     string endpoint_name = 3;
 }
 
-message DeleteNICInternalResponse {}
\ No newline at end of file
+message DeleteNICInternalResponse {}
+
+message IovEndpointPolicyInternalSetting {
```

Comment:
I think we talked about using a more abstract definition here, rather than encoding exactly what the HNS policy represents. Did we determine it's better this way?

Reply:
I had made a comment to remove the endpointpolicy portion of the name here, but the settings themselves I think are fine to leave in. I'm not sure how to make this more abstract though, if these are the settings we offer to be configurable on the adapter.

Reply:
I think I wasn't clear in what I'm asking about. :) I understand the data we pass to HNS has to fit a certain schema, and we can't change that. I'm instead asking why we are exposing that same schema to the next layer up here.

e.g. if the only case we ever care about is "use iov" vs "don't use iov", then we could expose an interface with a single `bool use_iov` setting, and then pass the correct data to HNS based on that.

Note I'm not saying we should do this, but I'm asking us to consider carefully what interface we expose, rather than default to leaking the lower level abstractions.

Does that make sense?

Reply:
@kevpar Yes, didn't realize the question was rooted in exposing all of the configurable settings sorry. Last we'd talked to the vnet folks they didn't have a strong opinion/they didn't even start thinking about what they would like configurable here so we leaned on the side of just exposing all of it. Maybe the better approach is the more restrained option(s) and we can expose more if there's a need for it.

Reply:
This issue here is that we don't know if the caller cares about what the other fields are ever. So the options are to either have a default value for queue pairs and interrupt moderation that we always enforce (which we don't know if the networking team cares about or not) or enforce that all three values are passed in for modification.

Reply:
My general feedback in this case would be we should understand the requirements of our callers when exposing an interface. Otherwise it's hard to tell if you're building the right thing.

In this case I think it will be fine to leave this as is, though. My point is more about what things we should consider when building abstractions.

Reply:
I see. I agree with you. It would be nice to only need to treat IOV as a bool :(

Reply:
Resolving for now